go/Makefile: Fix for go 1.8

sjackman opened the pull request:
Fix the error:
```
link: -X flag requires argument of the form importpath.name=value
```

adam-azarchs commented:
This change has conflicts with the fix that was adopted in https://github.com/10XGenomics/lariat/commit/396afce016873dc634641d3426e61896ce219d9d.  Of course, that one doesn't work any more with go 1.10+ because version strings don't sort.  If you rebase this change I will merge it.

sjackman committed:
go/Makefile: Fix for go 1.8

Fix the error:
link: -X flag requires argument of the form importpath.name=value

sjackman commented:
Is removing `LINK_SEPERATOR` what you had in mind?

adam-azarchs reviewed (approved):
Thanks.  Merging.

sjackman commented:
Thanks for merging, Adam!
